endpoints/openrtb2/amp_auction.go

```diff
@@ -465,6 +469,23 @@ func (deps *endpointDeps) overrideWithParams(ampParams amp.Params, req *openrtb2
 	return nil
 }
 
+// setConsentedProviders sets, if supplied, the addtl_consent value to user.ext.ConsentedProvidersSettings.consented_providers
+func setConsentedProviders(req *openrtb2.BidRequest, ampParams amp.Params) error {
+	if len(ampParams.AdditionalConsent) > 0 {
+		reqWrap := &openrtb_ext.RequestWrapper{BidRequest: req}
+
+		if userExt, err := reqWrap.GetUserExt(); err == nil {
+			userExt.SetConsentedProviders(ampParams.AdditionalConsent)
+		} else {
+			return err
+		}
+		if err := reqWrap.RebuildRequest(); err != nil {
+			return err
+		}
```

Comment:
Is there a way to add tests to trigger these errors?

Reply:
Done. Added two additional test cases.